File: app/schemas/generation.py
from typing import Optional

from pydantic import BaseModel, Field


class GenerateDraftRequest(BaseModel):
    category_id: Optional[int] = None
    category_name: Optional[str] = Field(default=None, max_length=200)
    extra_instruction: Optional[str] = Field(default=None, max_length=2000)
    organization_id: Optional[int] = None
    idempotency_key: Optional[str] = Field(default=None, min_length=8, max_length=128)


class GenerationJobResponse(BaseModel):
    id: int
    status: str
    content_id: Optional[int] = None
    generated_by_ai: bool = True
    title: Optional[str] = None
    body: Optional[str] = None
    error_code: Optional[str] = None
    error_message: Optional[str] = None

    model_config = {"from_attributes": True}

File: frontend/src/api/generation.ts
import { apiGet, apiPost } from './client';

export interface GeneratedDraft {

  id: number;
  title: string;
  body: string;
  status: string;
  organization_id?: number | null;
  generated_by_ai: boolean;
  generation_job_id?: number | null;
  media_id?: number | null;
}

export interface GenerateDraftRequest {
  category_id?: number;
  category_name?: string;
  extra_instruction?: string;
  organization_id?: number;
  idempotency_key?: string;
}

export function generateDraft(data: GenerateDraftRequest): Promise<GeneratedDraft> {
  return apiPost<GeneratedDraft>('generation/draft', data);
}


export interface AIProviderStatus {
  provider: string;
  model: string | null;
  configured: boolean;
  free_model: boolean;
  fallback_enabled: boolean;
  error?: string;
}

export function getAIProviderStatus(): Promise<AIProviderStatus> {
  return apiGet<AIProviderStatus>('generation/provider');
}
